### PR TITLE
changed: rename to opm-common

### PR DESCRIPTION
**Pull request by akva2**
This renames the module to opm-common.

### CODE REVIEW

**Comment by andlaus** [comment]
I guess the configure script (for dunecontrol compatibility) needs some changes, too?

**Comment by akva2** [comment]
grml. hard to keep track of all the hacks.

**Comment by akva2** [comment]
just a comment that needed changing, that was what you meant i guess.

**Comment by atgeirr** [comment]
I also noticed one thing in the CMakeLists.txt files (applies [unfortunately] to all modules): the message section (lines 43-55 in most) mentions opm-cmake in both its sub-points, but should mention opm-common.

**Comment by akva2** [comment]
changed in all modules.

**Comment by atgeirr** [comment]
I would like to merge all of this now, but perhaps a more careful review is in order? I am know to be a little trigger-happy sometimes... Does anyone volunteer to test this (today preferrably)? If not, I'd say let's trust that it works and just do it.

**Comment by babrodtk** [comment]
I can check if it compiles, give me a moment and I'll report on results

**Comment by akva2** [comment]
this now changes OPM_CMAKE_ROOT to OPM_COMMON_ROOT.

it would be confusing to use OPM_CMAKE_ROOT when the module we look for is called OPM_COMMON. furthermore we would have to set OPM_CMAKE_ROOT manually, while OPM_COMMON_ROOT is already set by the cmake find macros (for the opm-common package).

the OPM_MACROS_ROOT variable is still intact. this is the only one used internally in the build system, the others are used to determine the value of this one.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,17 +1,76 @@
...
+   find_package(opm-common QUIET)
+endif()
 
-list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/Modules)
-include(OpmInit)
+if (opm-common_FOUND)
+   include(OpmInit)
+else()
+   if (NOT OPM_COMMON_ROOT AND SIBLING_SEARCH)
+      set(OPM_COMMON_ROOT ${PROJECT_SOURCE_DIR}/../opm-common)
```
Comment (babrodtk) [inline CMakeLists.txt]: Does this really make sense here (i.e., searching for opm-common from within opm-common)? I don't think all of this change should be here.

Comment (akva2) [inline CMakeLists.txt]: indeed. good point..

File: cmake/Modules/opm-core-prereqs.cmake [hunk 2]
```diff
@@ -37,6 +37,7 @@ set (opm-core_DEPS
 	# DUNE dependency
 	"dune-common"
 	"dune-istl"
+	"opm-common REQUIRED"
```
Comment (babrodtk) [inline cmake/Modules/opm-core-prereqs.cmake]: Adding this gives me 
```
-- opm-core prereqs: .;opm-common-NOTFOUND
CMake Error at /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:108 (message):
  Could NOT find opm-core (missing: opm-core_ALL_PREREQS)
Call Stack (most recent call first):
  /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:315 (_FPHSA_FAILURE_MESSAGE)
  /home/babrodtk/workspace/opm/build/opm-rename/opm-common/cmake/Modules/OpmPackage.cmake:358 (find_package_handle_standard_args)
  /home/babrodtk/workspace/opm/build/opm-rename/opm-common/cmake/Modules/Findopm-core.cmake:16 (find_opm_package)
  /home/babrodtk/workspace/opm/build/opm-rename/opm-common/cmake/Modules/OpmFind.cmake:147 (find_package)
  /home/babrodtk/workspace/opm/build/opm-rename/opm-common/cmake/Modules/OpmFind.cmake:211 (find_and_append_package_to)
  /home/babrodtk/workspace/opm/build/opm-rename/opm-common/cmake/Modules/OpmLibMain.cmake:104 (find_and_append_package_list_to)
  CMakeLists.txt:107 (include)
```
for every module. Any ideas on what's missing?

Comment (akva2) [inline cmake/Modules/opm-core-prereqs.cmake]: try with updated opm-common / opm-core pr's.

Comment (babrodtk) [inline cmake/Modules/opm-core-prereqs.cmake]: I've got the same error in dune-cornerpoint. Can you check if the fix you made can be applied there as well (also possibly autodiff, polymer, ...)?

Comment (akva2) [inline cmake/Modules/opm-core-prereqs.cmake]: On 08/09/15 14:54, André R. Brodtkorb wrote:
>
> In cmake/Modules/opm-core-prereqs.cmake 
> <https://github.com/OPM/opm-common/pull/35#discussion_r38921240>:
>
> > @@ -37,6 +37,7 @@ set (opm-core_DEPS
> >  	# DUNE dependency
> >  	"dune-common"
> >  	"dune-istl"
> > +	"opm-common REQUIRED"
>
> I've got the same error in dune-cornerpoint. Can you check if the fix 
> you made can be applied there as well (also possibly autodiff, 
> polymer, ...)?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/OPM/opm-common/pull/35/files#r38921240>.
>
done.

Comment (babrodtk) [inline cmake/Modules/opm-core-prereqs.cmake]: Still doesn't work for me, can you help me by checking that all relevant changes were pushed and that it works at your end?

Comment (akva2) [inline cmake/Modules/opm-core-prereqs.cmake]: On 08/09/15 15:09, André R. Brodtkorb wrote:
>
> In cmake/Modules/opm-core-prereqs.cmake 
> <https://github.com/OPM/opm-common/pull/35#discussion_r38922686>:
>
> > @@ -37,6 +37,7 @@ set (opm-core_DEPS
> >  	# DUNE dependency
> >  	"dune-common"
> >  	"dune-istl"
> > +	"opm-common REQUIRED"
>
> Still doesn't work for me, can you help me by checking that all 
> relevant changes were pushed and that it works at your end?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/OPM/opm-common/pull/35/files#r38922686>.
>
my bad, i had forgotten a hunk.

we cannot do REQUIRED on the opm-common dependency right now as 
opm-common does not produce a library and all dependencies are assumed 
to contain one.

File: CMakeLists.txt (hunk 1 above)
Comment (andlaus) [inline CMakeLists.txt]: the CMakeLists.txt was copied from opm-core. I did not really go into semantic mode when adding it. ;)

File: cmake/Modules/opm-core-prereqs.cmake (hunk 2 above)
Comment (andlaus) [inline cmake/Modules/opm-core-prereqs.cmake]: > we cannot do REQUIRED on the opm-common dependency right now as opm-common does not produce a library and all dependencies are assumed to contain one.

that's strange: opm-material also does not produce a library and ewoms requires via REQUIRED. Maybe a test needs to be defined instead?

Comment (akva2) [inline cmake/Modules/opm-core-prereqs.cmake]: On 08/09/15 22:02, Andreas Lauser wrote:
>
> In cmake/Modules/opm-core-prereqs.cmake 
> <https://github.com/OPM/opm-common/pull/35#discussion_r38972584>:
>
> > @@ -37,6 +37,7 @@ set (opm-core_DEPS
> >  	# DUNE dependency
> >  	"dune-common"
> >  	"dune-istl"
> > +	"opm-common REQUIRED"
>
>     we cannot do REQUIRED on the opm-common dependency right now as
>     opm-common does not produce a library and all dependencies are
>     assumed to contain one.
>
> that's strange: opm-material also does not produce a library and ewoms 
> requires via REQUIRED. Maybe a test needs to be defined instead?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/OPM/opm-common/pull/35/files#r38972584>.
>
sorry for the low precision level. i meant header files, not library. 
it's the include path that holds the NOTFOUND value.

**Comment by babrodtk** [comment]
This now builds on my setup, so should (hopefully) be safe to start the merge-sync

**Comment by atgeirr** [comment]
So @akva2 and @andlaus: is everything ready for the big "merge all the PRs" operation?

**Comment by akva2** [comment]
yes.

**Comment by andlaus** [comment]
yes, too. (as far as I can tell)

**Pull request merged**